sormas-backend/src/main/java/de/symeda/sormas/backend/util/PatchHelper.java

```diff
@@ -0,0 +1,330 @@
+package de.symeda.sormas.backend.util;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import javax.validation.constraints.NotNull;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import de.symeda.sormas.api.EntityDto;
+import de.symeda.sormas.api.HasUuid;
+import de.symeda.sormas.api.utils.Experimental;
+
+public class PatchHelper {
+
+	private static ObjectMapper objectMapper = new ObjectMapper();
+	private static Map<Class, List<Field>> classFieldsMap = new HashMap<Class, List<Field>>();
+
+	/**
+	 * This method is used to partially update an existing object. It updates an existingObject with the values received in the jsonObject.
+	 * The existingObject has to be an object that contains a field UUID - thus an object instance of a class that implements HasUuid
+	 * interface
+	 * The jsonObject has to contain a field uuid and the existingObject uuid has to match with the uuid field in json.
+	 * The jsonObject can contains only the fields that are intended to be updated and the uuid of the object.
+	 * If the existingObject contains inner objects:
+	 * - a) of type HasUuid - those can be either updated or replaced.
+	 * In order to be updated the uuid of the inner object has to be provided and has to be the same as the existing one.
+	 * If the uuid differs the inner object is replaced.
+	 * - b) not of type HasUuid - are replaced.
+	 * !The logic if any inner object or value is allowed to be replaced or not relies entirely on the application!
+	 * 
+	 * THIS METHOD IS EXPERIMENTAL!!!!
+	 * 
+	 * @param jsonObject
+	 *            - the json payload containing the fields that are to be replaced. The structure of the json matches the structure of the
+	 *            Dto given by the existingObject
+	 * @param existingObject
+	 *            - the existing object whose fields are to be updated
+	 * @param <T>
+	 *            - the type of the object
+	 */
+	@Experimental
+	public static <T extends HasUuid> void postUpdate(@NotNull JsonNode jsonObject, @NotNull T existingObject) {
+
+		if (!existingObject.getUuid().equals(jsonObject.get(EntityDto.UUID).textValue())) {
+			throw new RuntimeException("The updated object must have a Uuid");
```

Comment:
please use ValidationRuntimeException with a translated message instead of RuntimeException